README.markdown

```diff
@@ -1,15 +1,31 @@
-recursion-schemes
-==========
+# recursion-schemes
 
 [![Hackage](https://img.shields.io/hackage/v/recursion-schemes.svg)](https://hackage.haskell.org/package/recursion-schemes) [![Build Status](https://secure.travis-ci.org/ekmett/recursion-schemes.png?branch=master)](http://travis-ci.org/ekmett/recursion-schemes)
 
-This package includes code for functional programming with bananas, envelopes, lenses, barbed wire and all that.
+## What is a recursion scheme?
 
-Contact Information
--------------------
+Many recursive functions share the same structure, e.g. pattern-match on the input and, depending on the data constructor, either recur on a smaller input or terminate the recursion with the base case. Another one: start with a seed value, use it to produce the first element of an infinite list, and recur on a modified seed in order to produce the rest of the list. Such a structure is called a recursion scheme.
 
-Contributions and bug reports are welcome!
```

Comment:
I think this and following message are good to leave in place still

Reply:
I don't hang out of irc though, should I keep that part anyway, with the reasonable assumption that readers will nevertheless be able to find a recursion-schemes fan there to answer their questions?

Reply:
I'm there, so is Edward :)